contracts/core/SetToken.sol

```diff
@@ -61,13 +63,24 @@ contract SetToken is
     // Mapping of componentHash to isComponent
     mapping(bytes32 => bool) internal isComponent;
 
+    // Address of the Factory contract that created the SetToken
+    address public factory;
+
     /* ============ Modifiers ============ */
 
     modifier isMultipleOfNaturalUnit(uint _quantity) {
         require((_quantity % naturalUnit) == 0);
         _;
     }
 
+    modifier isAuthorized() {
+        require(
```

Comment:
Maybe add a comment that only core is allowed. Or change the modifier name something like `isCore`